Chinese_Remainder_Theorem/Readme.md

```diff
@@ -0,0 +1,108 @@
+# Chinese Remainder Theorem
+
+If m<sub>1</sub>, m<sub>2</sub>, .., m<sub>k</sub> are pairwise relatively prime positive integers, and if a<sub>1</sub>, a<sub>2</sub>, .., a<sub>k</sub> are any integers, then the simultaneous congruences x ≡ a<sub>1</sub> (mod m<sub>1</sub>), x ≡ a<sub>2</sub> (mod m<sub>2</sub>), ..., x ≡ a<sub>k</sub> (mod m<sub>k</sub>) have a solution, and the solution is unique modulo m, where m = m<sub>1</sub>m<sub>2</sub>⋅⋅⋅m<sub>k</sub> .
+
+**Problem Statement :** 
+
+We are given two arrays num[0..n-1] and rem[0..n-1]. In num[0..n-1], every pair is coprime (gcd for every pair is 1). We need to find minimum positive number x such that:
+
+          x % num[0]  =  rem[0],
+          x % num[1]  =  rem[1],
+          x % num[2]  =  rem[2],
+          .......................
+          x % num[n-1]  =  rem[n-1]
+          
+## Algorithm
+
+The solution is based on the below given formula :
+
+        x =  sum of { (rem[i]* pp[i]* inv[i]) % prod } where 0 <= i <= n-1
+
+        where,
+        
+        rem[i] is given array of remainders
+
+        prod is product of all given numbers
+        prod = num[0] * num[1] * ... * num[k-1]
+
+        pp[i] is product of all divided by num[i]
+        pp[i] = prod / num[i]
+
+        inv[i] = Modular Multiplicative Inverse of pp[i] with respect to num[i] 
+
+
+## Example
+
+![Pic01](https://i.pinimg.com/474x/f4/62/23/f46223a470eb8f967b3be60f8a255c6e--chinese-remainder-theorem-the-chinese.jpg)
```

Comment:
Since this image has been taken from internet. Please mention the source from this in references.